feed-rs/src/model.rs

```diff
@@ -984,6 +999,351 @@ impl Person {
     }
 }
 
+/// Represents a "podcast:person" from the [Podcast spec].
+///
+/// [Podcast spec]: https://podcastindex.org/namespace/1.0#person
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PodcastPerson {
+    pub name: String,
+    #[serde(deserialize_with = "podcast_enum")]
+    pub role: Role,
+    #[serde(deserialize_with = "podcast_enum")]
+    pub group: Group,
+    pub img: Option<Url>,
+    pub href: Option<Url>,
+}
+
+/// Represents a [`PodcastPerson`](PodcastPerson)'s role.
+///
+/// See the [taxonomy](https://github.com/Podcastindex-org/podcast-namespace/blob/main/taxonomy.json)
+/// for a list of roles and their associated groups.
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub enum Role {
+    Director,
+    #[serde(rename = "Assistant Director")]
+    AssistantDirector,
+    #[serde(rename = "Executive Producer")]
+    ExecutiveProducer,
+    #[serde(rename = "Senior Producer")]
+    SeniorProducer,
+    Producer,
+    #[serde(rename = "Associate Producer")]
+    AssociateProducer,
```

Comment:
I tossed up between using an enum or a `String` for a while, but since the role taxonomy is a fixed list with a [canonical set](https://github.com/Podcastindex-org/podcast-namespace/blob/main/taxonomy.json), I decided an enum would be best.

It's unfortunate that `serde` doesn't offer something along the lines of `#[rename = "Title Case"]` out of the box. I could have brought in an [external dependency](https://docs.rs/heck/latest/heck/struct.AsTitleCase.html) or [two](https://crates.io/crates/strum), but decided against it. `feed-rs` has a pretty minimal set of dependencies, and I didn't want to add more unnecessarily.